### ISSUE TITLE
No login after update to SFOS 4.4.0.58

### ISSUE THREAD

**Issue by broncheolus** [sailfish-spotify/hutspot#119]
SFOS: 4.4.0.58
device: XA2

After update to SFOS 4.4.0.58 I had to login to hutspot again. This was not possible due to a white login page and due to a not working login over the standard browser of SF (see attached screenshot). I really hope that there is a workaround possible or an "easy" fix for you because I´m a huge fan of hutspot and would really miss it on my device. Thanks for your great work for SF. Greetings from Austria, broncheolus.

edit: I managed to reregister when starting hutspot from commandline (login page shows). After that hutspot works from commandline but still not from icon. I will investigate further.

edit: all the config folders seem to be empty (in .config and .local/share)

edit: Oh, is it the thing with sailjail disable and the .desktop file? One has to edit it, I think. But not knowing how, so next step to be solved. Just a little help would be appreciated. Thx.

![photo_2022-04-03_16-57-59](https://user-images.githubusercontent.com/25463978/161434234-b40b796b-f7f0-4305-af87-195d8a27c853.jpg)

**Comment by poetaster** [comment]
I think the fix is in the last PR from @attah where he moves to a the new WebView. How had you installed?

**Comment by broncheolus** [comment]
Thanks for your answer. I installed it from Openrepos. I just cannot remember if I used storeman or if I downloaded and installed  it manually.

**Comment by poetaster** [comment]
Hmmm. Ok, I've only updated the fixed version, with ssl 1.1 support and the new web view to chum. You can use the chum gui  (install with https://chumrpm.netlify.app/) or download the rpm at: https://api.merproject.org/package/show/sailfishos:chum/hutspot

**Comment by broncheolus** [comment]
Sorry for the late reply. I installed hutspot from chum (I finally found it there, it´s not listed as application so one has to switch to "show all packages")

Now the web login page opens but after being told to return to hutspot from "login page" no device can be found.
Starting hutspot from console works again with the standard login screen. Only the device has to be set as "?recent device?" (I just know the german words) every time after starting hutspot. 

Just for your knowledge: Hutspot shows version 0.2 in about although 0.4 from chum is installed.

Edit: I now tried different variants for login, external browser, normal login, from console, from icon. Sometimes the login from icon start had an influence on hutspot started from console. (i.e. that the device is active directly after startup from console) All in all very confusing for me because of all the different results. But in the end I was always able to activate the device in different ways.  

But I´m sorry, I cannot write it all down so it´s  of any help for you. It´s too confusing, sorry.

I really appreciate your work for hutspot. Keep it up. Thanks and greetings from Austria.

**Comment by poetaster** [comment]
> Now the web login page opens but after being told to return to hutspot from "login page" no device can be found.
> Starting hutspot from console works again with the standard login screen. Only the device has to be set as "?recent device?" (I just know the german words) every time after starting hutspot.

Ok, it's difficult for me to test, but the patch came from @attah and he might be able to help me narrow it down. 

> Just for your knowledge: Hutspot shows version 0.2 in about although 0.4 from chum is installed.

That was hardcoded in the QML. I'll go fix that.

> But I´m sorry, I cannot write it all down so it´s of any help for you. It´s too confusing, sorry.

You can also write me in German if that helps. PM on the forum (also poetaster there).

**Comment by attah** [comment]
> no device can be found.

Well, do you have any?
Hutspot in and of itself is not a playback "device".
The intended use is to also have LibreSpot running afaik - i just tested with having Spotify running on my desktop and controlling that from Hutspot.

**Comment by broncheolus** [comment]
Sorry for not writing back to you. Well German would be easier but I think English will also do. I did not use hutspot lately but came back to this thread today. What I did not realise is that when starting hutspot **from icon librespot cannot be found**. From terminal it gets  found. It´s already started too, the only thing that has to be done is to choose the device as "aktuelles Gerät".
Thank you. I hope I´m of any help.

**Comment by attah** [comment]
So logging did in fact work.
Then we at least know what to look for: jailed Hutspot not detecting Librespot.
That it can no longer be started/stopped from there is kinda expected, but wonder what could break detection...

**Comment by Scarrough** [comment]
Might this be handled by modifying the permissions?

**Comment by poetaster** [comment]
> Might this be handled by modifying the permissions?

I don't think so. librespot as a binary is not supported (ffmpeg, for instance, is). But it's probably worth a shot
```
[X-Sailjail]
Sandboxing=Disabled
```
in the .desktop file at the end.

**Comment by Scarrough** [comment]
> > Might this be handled by modifying the permissions?
> 
> I don't think so. librespot as a binary is not supported (ffmpeg, for instance, is). But it's probably worth a shot
> 
> ```
> [X-Sailjail]
> Sandboxing=Disabled
> ```
> 
> in the .desktop file at the end.

Yep, that did the trick.

**Comment by broncheolus** [comment]
Thank you, disabling sandboxing indeed helped. The only thing that has to be done after that, is choosing the inactive device as "aktuelles Gerät". But that´s the least problem. Thank you for your effort. Keep on sailing.
